pyang/statements.py

```diff
@@ -2193,45 +2245,42 @@ def v_reference_deviate(ctx, stmt):
                     t.substmts.append(c)
     elif stmt.arg == 'replace':
         for c in stmt.substmts:
+            if c.keyword == '_comment':
+                continue
             if c.keyword == 'config' and hasattr(t, 'i_config'):
                 # config is special: since it is an inherited property
-                # with a default, all nodes has a config property.  this means
-                # that it can only be replaced.
-                # first, set the property...
-                if c.arg == 'true':
-                    t.i_config = True
-                elif c.arg == 'false':
-                    t.i_config = False
-                # ... and then modify the original statement, if any
+                # with a default, all nodes has a config property.
+                # first, save the old property, and then set the property...
+
                 old = t.search_one(c.keyword)
                 if old is not None:
+                    negc = copy.copy(old)
                     old.arg = c.arg
                 else:
-                    t.substmts.append(c)
-                # make sure the target's children have proper config stmts
-                sub = t.substmts
-                if hasattr(t, 'i_children'):
-                    sub = sub + t.i_children
-                for d in sub:
-                    if d.keyword in data_definition_keywords:
-                        if hasattr(d, 'i_config') and d.i_config != t.i_config:
-                            # this child has another config property,
-                            # maybe fix the statment
-                            old = d.search_one('config')
-                            if old is None:
-                                negc = copy.copy(c)
-                                if c.arg == 'true':
-                                    negc.arg = 'false'
-                                else:
-                                    negc.arg = 'true'
-                                d.substmts.append(negc)
+                    err_add(ctx.errors, t.pos, 'BAD_DEVIATE_REP', (c.keyword, t.i_module.arg, t.arg))
```

Comment:
Please have a look at CONTRIBUTING; there are just two rules ;-)

Reply:
It's my fault. Didn't notice it's more than 80 characters.